include/vast/Util/Warnings.hpp

```diff
@@ -47,16 +47,54 @@ VAST_RELAX_WARNINGS
 #include <llvm/Support/Debug.h>
 VAST_UNRELAX_WARNINGS
 
+#include <stdexcept>
+
 #define DEBUG_TYPE "vast"
 
+#ifdef ENABLE_VAST_EXCEPTION
+  #define vast_error vast::error<std::runtime_error>
+#else
+  #define vast_error llvm::dbgs
+#endif
+
+#define vast_debug llvm::dbgs
+
 namespace vast
 {
+#ifdef ENABLE_VAST_EXCEPTION
+    template <typename T = std::runtime_error>
+    struct error {
+      private:
+        std::string msg;
+        bool triggered = false;
```

Comment:
I am confused about the name, because if we look in the `dtor`, it only throws `if triggered is true`. So either rename it or fix the values.

Reply:
Removed `triggered` flag from the `error` class. My initial thought was to enable the exception only if `triggered` is set to true. But probably we don't need to have all these complexity and should have simple implementation of `error_stream`.